plugins/modules/fault_set.py

```diff
@@ -0,0 +1,400 @@
+'''
+
+RETURN = r'''
+changed:
+    description: Whether or not the resource has changed.
+    returned: always
+    type: bool
+    sample: 'false'
+
+fault_set_details:
+    description: Details of fault set.
+    returned: always
+    type: dict
+    contains:
+        protectionDomainId:
+            description: The ID of the protection domain.
+            type: str
+        name:
+            description: device name.
+            type: str
+        id:
+            description: device id.
+            type: str
+        links:
+            description: Fault set links.
+            type: list
+            contains:
+                href:
+                    description: Fault Set instance URL.
+                    type: str
+                rel:
+                    description: Relationship of fault set with different
+                                 entities.
+                    type: str
+    sample: {
+        "protectionDomainId": "da721a8300000000",
+        "name": "fs_001",
+        "id": "eb44b70500000000",
+        "links": [
+            { "rel": "self", "href": "/api/instances/FaultSet::eb44b70500000000" },
+            {
+                "rel": "/api/FaultSet/relationship/Statistics",
+                "href": "/api/instances/FaultSet::eb44b70500000000/relationships/Statistics"
+            },
+            {
+                "rel": "/api/FaultSet/relationship/Sds",
+                "href": "/api/instances/FaultSet::eb44b70500000000/relationships/Sds"
+            },
+            {
+                "rel": "/api/parent/relationship/protectionDomainId",
+                "href": "/api/instances/ProtectionDomain::da721a8300000000"
+            }
+        ]
+        }
+
+'''
+
+
+from ansible_collections.dellemc.powerflex.plugins.module_utils.storage.dell import (
+    utils,
+)
+from ansible.module_utils.basic import AnsibleModule
+
+
+LOG = utils.get_logger("fault_set")
+
+
+class PowerFlexFaultSet(object):
+    """Class with FaultSet operations"""
+
+    def __init__(self):
+        """Define all parameters required by this module"""
+        self.module_params = utils.get_powerflex_gateway_host_parameters()
+        self.module_params.update(get_powerflex_fault_set_parameters())
+
+        mutually_exclusive = [
+            ["fault_set_name", "fault_set_id"],
+            ["protection_domain_name", "protection_domain_id"],
+        ]
+
+        # Weird issue with require_if if only on option was tied to the requirement
+        # so added dummy zz option to the list as workaround
+        required_if = [
+            ("state", "present", ("protection_domain_id",
+             "protection_domain_name"), True),
+            ("state", "present", ("fault_set_name", "zz"), True),
+            ("state", "absent", ("fault_set_id", "zz"), True),
+        ]
+
+        # initialize the Ansible module
+        self.module = AnsibleModule(
+            argument_spec=self.module_params,
+            supports_check_mode=False,
```

Comment:
As we have to support the check mode.
```suggestion
            supports_check_mode=True,
```